File: 1823-find-the-winner-of-the-circular-game/main.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        vector<int> friends;
        for (int i = 1; i <= n; ++i) {
            friends.push_back(i);
        }

        int idx = 0;

        while (friends.size() > 1) {
            idx = (idx + k - 1) % friends.size();
            friends.erase(friends.begin() + idx);
        }

        return friends[0];
    }
};

File: 1823-find-the-winner-of-the-circular-game/main.rs
impl Solution {
    pub fn find_the_winner(n: i32, k: i32) -> i32 {
        let mut friends: Vec<i32> = (1..=n).collect();
        let mut idx: usize = 0;

        while friends.len() > 1 {
            idx = (idx + k as usize - 1) % friends.len();
            friends.remove(idx);
        }

        friends[0]
    }
}